packages/headless/src/features/configuration/configuration-state.ts

```diff
@@ -18,35 +18,22 @@ export interface ConfigurationState {
    */
   accessToken: string;
   /**
-   * The Platform URL to use.
-   * By default, https://platform.cloud.coveo.com
+   * The base platform [organization endpoint](https://docs.coveo.com/en/mcc80216).
+   *
+   * This value is automatically resolved from the `organizationId` and `environment`.
+   *
+   * For example, if `organizationId` is `mycoveocloudorganizationg8tp8wu3` and `environment` is `prod`, `platformUrl`
+   * will be `https://mycoveocloudorganizationg8tp8wu3.org.coveo.com`.
    */
   platformUrl: string;
   /**
    * The global headless engine Search API configuration.
    */
-  search: {
-    /**
-     * The Search API base URL to use.
-     * By default, will append /rest/search/v2 to the platformUrl value.
-     */
-    apiBaseUrl: string;
-    /**
-     * The locale of the current user. Must comply with IETF’s BCP 47 definition: https://www.rfc-editor.org/rfc/bcp/bcp47.txt.
-     */
-    locale: string;
-    /**
-     * The [tz database](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) identifier of the time zone to use to correctly interpret dates in the query expression, facets, and result items.
-     * By default, the timezone will be [guessed](https://day.js.org/docs/en/timezone/guessing-user-timezone).
-     */
-    timezone: string;
-    /**
-     * Specifies the name of the authentication providers to use to perform queries.
-     *
-     * See [SAML Authentication](https://docs.coveo.com/en/91/).
-     */
-    authenticationProviders: string[];
-  };
+  search: SearchState;
+  /**
+   * The global headless engine Commerce API configuration.
+   */
+  commerce: CommerceState;
```

Comment:
Why is the commerce state bleeding all over into the Base search stuff?

Doesn't make sense to have commerce state in the Configuration State.

There should be a CommerceConfigurationState which is a new type that unions ConfigurationState and Commerce state no?
That type should only be used in the Commerce specific situations.